validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -139,6 +141,10 @@ private Map<String, Object> getActualLog(
       dependencyFilter = String.format("&& ($.RemoteService = \"%s\") && ($.RemoteOperation = \"%s\")", remoteService, remoteOperation);
     }
 
+    if (remoteResourceType != null && remoteResourceIdentifier != null) {
+      dependencyFilter += String.format(" && ($.RemoteResourceType = %%%s%%) && ($.RemoteResourceIdentifier = %%%s%%)", remoteResourceType, remoteResourceIdentifier);
```

Comment:
If we expect to be searching with the value directly, we can likely just remove the `^...$` RegEx symbols in the validation templates and remove the need for RegEx here altogether. Not a 100% necessary change but doing so will improve the search time by filtering for the exact string instead of a RegEx pattern.

Reply:
Hey Mahad, we shouldn't expect the same value for either remoteResourceType or remoteResourceIdentifier. You can verify this in the following run: [GitHub Workflow Run](https://github.com/ektabj/aws-application-signals-test-framework/actions/runs/10580614526/job/29315930502).

1. The /aws-sdk-call API shows different values.
2. The /mysql API also has different values.

I believe these values can't be static since some of them are concatenated into a string only during runtime. Let me know if I've missed anything.

Reply:
Hi @ektabj 

I think Mahad suggests to not use regex here.

As you can see, we are not using regex for RemoteResourceType in the template.
For RemoteResourceIdentifier, we can change the value from `^{{remoteResourceIdentifier}}$` to `{{remoteResourceIdentifier}}` in the template.

Template: https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/java/eks/rds-mysql-log.mustache#L34-L35